assets/js/src/form_editor/blocks/custom_checkbox/custom_field_settings.jsx

```diff
@@ -15,38 +15,51 @@ const CustomFieldSettings = ({
   checkboxLabel,
   isDeleting,
   onCustomFieldDelete,
+  onChange,
 }) => {
   const [localMandatory, setLocalMandatory] = useState(mandatory);
   const [localIsChecked, setLocalIsChecked] = useState(isChecked);
   const [localCheckboxLabel, setLocalCheckboxLabel] = useState(checkboxLabel);
 
+  const localData = useMemo(() => ({
```

Comment:
useMemo is here so that the localData is computed only when some of local state properties changes. localData are used as a dependency for an effect which triggers onChange callback.
Each time localData changes it triggers onChange callback so that is why it has to be memoized.